core/services/workflows/metering/metering.go

```diff
@@ -177,48 +175,85 @@ func (r *Report) Deduct(ref string, amount int64) error {
 	}
 
 	r.steps[ref] = ReportStep{
-		Deduction: amount,
+		Deduction: amount.Decimal,
 		Spends:    nil,
 	}
 
 	// if in metering mode, exit early without modifying local balance
-	if r.balance.allowNegative {
+	if r.balance.meteringMode || !amount.Valid {
 		return nil
 	}
 
-	err := r.balance.Minus(amount)
+	err := r.balance.Minus(amount.Decimal)
 	if err != nil {
 		return err
 	}
 
 	return nil
 }
 
-// GetAvailableForInvocation returns the amount of credits that can be used based on the available credit balance.
-// This is determined by dividing unearmarked local credit balance by the number of potential concurrent calls.
-func (r *Report) GetAvailableForInvocation(openConcurrentCallSlots int) (int64, error) {
+func (r *Report) CreditToSpendingLimits(
+	info capabilities.CapabilityInfo,
+	amount decimal.Decimal,
+) []capabilities.SpendLimit {
+	if len(info.SpendTypes) > 0 {
+		spendType := info.SpendTypes[0]
+
+		// use rate card to convert capSpendLimit to native units
+		spendLimit := r.balance.ConvertFromBalance(string(spendType), amount)
+
+		// check to see if metering mode is active
+		if r.balance.meteringMode {
+			r.lggr.Errorf("no rate exists in rate card for %s; entering metering mode", spendType)
```

Comment:
This `if` block doesn't flip it into metering mode. This log will end up being spammy if it _is_ actually in metering mode.

How about just moving the `if r.balance.meteringMode` up to combine with `if len(info.SpendTypes) > 0`:
```
	if len(info.SpendTypes) > 0 && !r.balance.meteringMode {
...
```